### DISCUSSION TITLE
Quarkus TLS authentication throws 401 if certificate expires. How to update/change it to throw 5xx error code instead

### DISCUSSION THREAD

**Discussion by ngarg-panw** [keycloak/keycloak#26616] [Q&A]
Hi,

I have Certificate based authentication configured successfully. 

Here are the contents for docker file:
`xec java $JAVA_OPTS $APM_JAVA_OPTS "-Djavax.net.ssl.trustStore=${TRUSTSTORE_LOCATION}" "-Djavax.net.ssl.trustStorePassword=${TRUSTSTORE_PASSWORD}" '-Djavax.net.ssl.trustStoreType=jks' -jar app.jar`

Here are the contents from application.yml
`    http:
      ssl-port: 8080
      port: 0
      test-port: 0
      ssl:
        certificate:
          key-store-file: ${KEYSTORE_LOCATION}
          key-store-password: ${KEYSTORE_PASSWORD}
          key-store-file-type: JKS
`

in case the certificate is expired, the framework throws 401-UnAuthorized error. I want to change it to throw 5xx instead with appropriate error message. How can I achieve that?

**Comment by vmuzikar** [comment]
I'd suggest opening an issue in Quarkus as the TLS cert handling is transparent to Keycloak.

**Reply by ngarg-panw** [reply]
Thanks! Opened in quarkus https://github.com/quarkusio/quarkus/discussions/38515

**Comment by tsaarni** [comment]
It is unclear what your setup looks like. You mention TLS certificate authentication, but also that you receive HTTP response codes. If TLS authentication is failing, I would have expected no HTTP responses at all since connection is terminated before HTTP comes into play. From the configuration above you say`app.jar` is launched, and on the second line it might suggest you have Quarkus configuration in YAML format (with broken linefeeds). Do you possibly mean you have your own application, built on Quarkus? What is the relation to Keycloak? Maybe you mean that after your application fails to connect to Keycloak, it responds to its own client with `401`, possibly using something like [this](https://quarkus.io/guides/security-keycloak-authorization)?
